pkg/registry/admissionregistration/validatingadmissionpolicy/authz.go

```diff
@@ -59,16 +60,17 @@ func (v *validatingAdmissionPolicyStrategy) authorize(ctx context.Context, polic
 		return nil
 	}
 
-	// for superuser, skip all checks
-	if rbacregistry.EscalationAllowed(ctx) {
-		return nil
-	}
-
 	user, ok := genericapirequest.UserFrom(ctx)
 	if !ok {
 		return fmt.Errorf("cannot identify user to authorize read access to paramKind resources")
 	}
 
+	// for superuser, skip all checks
+	isSystemAdmin := sets.New(user.GetGroups()...).Has(userpkg.SystemPrivilegedGroup)
```

Comment:
Hmm… I sort of prefer the authorization check over checking a hardcoded group… do we do hardcoded group checks for any other authz purposes? I can't think of any

Reply:
what would it be like the authorization check? can you give an example?

Reply:
I think it looks like what the RBAC registries do:

```
	if rbacregistry.EscalationAllowed(ctx) || rbacregistry.RoleEscalationAuthorized(ctx, s.authorizer) {
```

They check for presence of a specific hard-coded superuser group *or* if the authorizer allows the operation using an `escalate` verb.

cc @deads2k for visibility as well

Reply:
@sttts @jpbetz

Reply:
My original comment was the inverse: using `EscalationAllowed` here is a misuse. We are not checking for escalation, but for being privileged.

Do we really check for `rbacregistry.RoleEscalationAuthorized` in such cases? This sounds wrong. Just because you have the right to escalate does not mean you want to use that right.

Reply:
I don't feel strongly about checking RBAC escalation (and using an RBAC method to check for admission permissions does seem weird, I agree).

*But* checking only a hard-coded group doesn't seem correct. We shouldn't require someone to use this specific group to have a privileged user, we should let the authorization system decide if they are privileged, and *one* way the authorizer could decide that is if they are a member of the hard-coded superuser group.

Reply:
The `RoleEscalationAuthorized` call was an example of using the authorizer to check permissions, I wasn't suggesting calling that method exactly

Reply:
Sorry, I'm lost,  I agree with everything. Where do I go now?